### PR TITLE
add a close event to the java connection library

### PR DESCRIPTION
**Pull request by devoncarew**
- add a close event to the java connection library
- commit IntelliJ metadata files for the java and dart directories

@danrubel

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,8 +18,7 @@ pubspec.lock
 *.md~
 
 # IDEs
-.idea
-*.iml
+.idea/
```
Comment (danrubel) [inline .gitignore]: If we want to include intelliJ metadata then why not remove this?

**Comment by danrubel** [comment]
LGTM once travis errors are addressed.

File: .gitignore (hunk 1 above)
Comment (devoncarew) [inline .gitignore]: There were a few files in there that we didn't want (some of the metadata is common for the project, and some is user and machine specific).

**Comment by devoncarew** [comment]
fixed!

**Pull request merged**